libraries/AP_BattMonitor/AP_BattMonitor_SMBus_Solo.cpp

```diff
@@ -39,12 +41,12 @@ AP_BattMonitor_SMBus_Solo::AP_BattMonitor_SMBus_Solo(AP_BattMonitor &mon,
 
 void AP_BattMonitor_SMBus_Solo::timer()
 {
-    uint8_t buff[8];
+    uint8_t buff[12];
     uint32_t tnow = AP_HAL::micros();
 
 
     // read cell voltages
-    if (read_block(BATTMONITOR_SMBUS_SOLO_CELL_VOLTAGE, buff, 8)) {
+    if (!_use_extended and read_block(BATTMONITOR_SMBUS_SOLO_CELL_VOLTAGE, buff, 8)) {
```

Comment:
```suggestion
    if (!_use_extended && read_block(BATTMONITOR_SMBUS_SOLO_CELL_VOLTAGE, buff, 8)) {
```
We don't typically use the full words.